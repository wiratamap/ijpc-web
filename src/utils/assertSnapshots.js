import React from 'react';
import renderer from 'react-test-renderer';

/**
 * @param Component - (Component) Component that needs to be test.
 * @param testConfiguration - (object) object of props and test description
 */
const assertSnapshot = (Component, testConfiguration) => {
  const { props, description } = testConfiguration;

  describe('#render', () => {
    it(description, () => {
      const component = renderer.create(<Component {...props} />).toJSON();

      expect(component).toMatchSnapshot();
    });
  });
};

const assertSnapshots = (Component, testConfigurations) => {
  testConfigurations.forEach((testConfiguration) => assertSnapshot(Component, testConfiguration));
};

export default assertSnapshots;
