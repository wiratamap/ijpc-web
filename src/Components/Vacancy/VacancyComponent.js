import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

import { currencyFormat } from '../../utils/numberFormat';
import placeholderImage from '../../assets/placeholder-image-768x576.png';

const VacancyComponent = ({ vacancy }) => (
  <div className="card">
    <ul className="list-group list-group-flush">
      <li className="list-group-item">
        <div className="row m-0">
          <div className="col-2">
            <img
              src={placeholderImage}
              className="img-thumbnail image-thumbnail-item"
              alt="1"
            />
          </div>
          <div className="col-6">
            <h4>{vacancy.subject}</h4>
            <p className="font-weight-light">{vacancy.employer}</p>
            <p className="font-weight-light text-small">
              {vacancy.location}
              {' '}
              ·
              {' '}
              {currencyFormat(vacancy.salaryCurrency, vacancy.approximateMinimumSalary)}
              {' '}
              -
              {' '}
              {currencyFormat(vacancy.salaryCurrency, vacancy.approximateMaximumSalary)}
              {' '}
              /
              {' '}
              {vacancy.payrollInterval.name}
            </p>
          </div>
          <div className="col-4">
            <div className="row justify-content-end">
              <p className="font-weight-light text-small">
                Posted
                {' '}
                { moment(vacancy.createdAt).fromNow() }
                {' '}
                · Apply before
                {' '}
                { moment(vacancy.submissionDeadline).format('D MMM') }
                .
              </p>
            </div>
            <div className="row justify-content-end">
              <button type="button" className="btn btn-outline-primary">Vacancy Detail</button>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
);

VacancyComponent.propTypes = {
  vacancy: PropTypes.shape({
    subject: PropTypes.string,
    employer: PropTypes.string,
    location: PropTypes.string,
    salaryCurrency: PropTypes.string,
    approximateMinimumSalary: PropTypes.number,
    approximateMaximumSalary: PropTypes.number,
    payrollInterval: PropTypes.shape({
      name: PropTypes.string
    }),
    createdAt: PropTypes.string.isRequired,
    submissionDeadline: PropTypes.string.isRequired
  })
};

VacancyComponent.defaultProps = {
  vacancy: {
    subject: '',
    employer: '',
    location: '',
    salaryCurrency: 'IDR',
    approximateMinimumSalary: 0,
    approximateMaximumSalary: 0,
    payrollInterval: {
      name: 'month'
    }
  }
};

export default VacancyComponent;
