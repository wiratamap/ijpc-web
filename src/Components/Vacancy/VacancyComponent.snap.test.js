import { VACANCIES } from '../../fixtures/vacanciesData';
import assertSnapshots from '../../utils/assertSnapshots';
import VacancyComponent from './VacancyComponent';

jest.mock('../../utils/numberFormat', () => ({
  currencyFormat: () => 'IDR 0.00'
}));

describe('VacancyComponent', () => {
  const configurations = [
    {
      props: {
        vacancy: {
          subject: '',
          employer: '',
          location: '',
          salaryCurrency: 'IDR',
          approximateMinimumSalary: 0,
          approximateMaximumSalary: 0,
          payrollInterval: {
            name: 'month'
          },
          createdAt: VACANCIES.content[0].createdAt,
          submissionDeadline: VACANCIES.content[0].submissionDeadline
        }
      },
      description: 'should render VacancyComponent with defined posted date, deadline, and another default props'
    },
    {
      props: {
        vacancy: {
          subject: VACANCIES.content[0].subject,
          employer: VACANCIES.content[0].employer,
          location: VACANCIES.content[0].location,
          salaryCurrency: VACANCIES.content[0].salaryCurrency,
          approximateMinimumSalary: VACANCIES.content[0].approximateMinimumSalary,
          approximateMaximumSalary: VACANCIES.content[0].approximateMaximumSalary,
          payrollInterval: VACANCIES.content[0].payrollInterval,
          createdAt: VACANCIES.content[0].createdAt,
          submissionDeadline: VACANCIES.content[0].submissionDeadline
        }
      },
      description: 'should render VacancyComponent with proper defined props'
    }
  ];

  assertSnapshots(VacancyComponent, configurations);
});
