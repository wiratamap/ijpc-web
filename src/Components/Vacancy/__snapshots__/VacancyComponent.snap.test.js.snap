// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VacancyComponent #render should render VacancyComponent with defined posted date, deadline, and another default props 1`] = `
<div
  className="card"
>
  <ul
    className="list-group list-group-flush"
  >
    <li
      className="list-group-item"
    >
      <div
        className="row m-0"
      >
        <div
          className="col-2"
        >
          <img
            alt="1"
            className="img-thumbnail image-thumbnail-item"
            src="placeholder-image-768x576.png"
          />
        </div>
        <div
          className="col-6"
        >
          <h4>
            
          </h4>
          <p
            className="font-weight-light"
          >
            
          </p>
          <p
            className="font-weight-light text-small"
          >
            
             
            路
             
            IDR 0.00
             
            -
             
            IDR 0.00
             
            /
             
            month
          </p>
        </div>
        <div
          className="col-4"
        >
          <div
            className="row justify-content-end"
          >
            <p
              className="font-weight-light text-small"
            >
              Posted
               
              2 days ago
               
              路 Apply before
               
              14 May
              .
            </p>
          </div>
          <div
            className="row justify-content-end"
          >
            <button
              className="btn btn-outline-primary"
              type="button"
            >
              Vacancy Detail
            </button>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`VacancyComponent #render should render VacancyComponent with proper defined props 1`] = `
<div
  className="card"
>
  <ul
    className="list-group list-group-flush"
  >
    <li
      className="list-group-item"
    >
      <div
        className="row m-0"
      >
        <div
          className="col-2"
        >
          <img
            alt="1"
            className="img-thumbnail image-thumbnail-item"
            src="placeholder-image-768x576.png"
          />
        </div>
        <div
          className="col-6"
        >
          <h4>
            VP of Engineering
          </h4>
          <p
            className="font-weight-light"
          >
            My Company Inc
          </p>
          <p
            className="font-weight-light text-small"
          >
            West Jakarta, Indonesia
             
            路
             
            IDR 0.00
             
            -
             
            IDR 0.00
             
            /
             
            month
          </p>
        </div>
        <div
          className="col-4"
        >
          <div
            className="row justify-content-end"
          >
            <p
              className="font-weight-light text-small"
            >
              Posted
               
              2 days ago
               
              路 Apply before
               
              14 May
              .
            </p>
          </div>
          <div
            className="row justify-content-end"
          >
            <button
              className="btn btn-outline-primary"
              type="button"
            >
              Vacancy Detail
            </button>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
